server/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -53,16 +89,43 @@ public long nowInMillis() {
     }
 
     /**
-     * Returns an instance of {@link SearchExecutionContext} if available of null otherwise
+     * Returns an instance of {@link SearchExecutionContext} if available or null otherwise
      */
     public SearchExecutionContext convertToSearchExecutionContext() {
         return null;
     }
 
+    /**
+     * Returns an instance of {@link CoordinatorRewriteContext} if available or null otherwise
+     */
     public CoordinatorRewriteContext convertToCoordinatorRewriteContext() {
         return null;
     }
 
+    public MappedFieldType getFieldType(String name) {
+        return failIfFieldMappingNotFound(name, fieldType(name));
+    }
+
+    protected MappedFieldType fieldType(String name) {
+        // If the field is not allowed, behave as if it is not mapped
+        if (allowedFields != null && false == allowedFields.test(name)) {
+            return null;
+        }
+        MappedFieldType fieldType = runtimeMappings.get(name);
+        return fieldType == null ? mappingLookup.getFieldType(name) : fieldType;
+    }
+
+    public IndexAnalyzers getIndexAnalyzers() {
+        if (mapperService == null) {
+            return IndexAnalyzers.of(Collections.emptyMap());
+        }
+        return mapperService.getIndexAnalyzers();
+    }
+
+    MappedFieldType failIfFieldMappingNotFound(String name, MappedFieldType fieldMapping) {
```

Comment:
This method is a bit confusingly named now, given that the default implementation *doesn't* fail if the mapping isn't found!